### PR TITLE
fix(tab-title, tab): Support nested tabs

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #4198

## Summary

fix(tab-title, tab): Support nested tabs. #4198

Also does some cleanup to related files.

### CODE REVIEW

File: src/components/tabs/tabs.e2e.ts [hunk 1]
```diff
@@ -261,4 +261,57 @@ describe("calcite-tabs", () => {
     expect(finalSelectedItem.titleTab).toBe("title-2");
     expect(finalSelectedItem.contentTab).toBe("tab-2");
   });
+
+  it("item selection should work with nested tabs", async () => {
+    const page = await newE2EPage({
+      // load page with the dropdown template,
```
Comment (jcfranco) [inline src/components/tabs/tabs.e2e.ts:267]: These comments seem unrelated.

File: src/components/tabs/tabs.e2e.ts [hunk 2]
```diff
@@ -261,4 +261,57 @@ describe("calcite-tabs", () => {
...
+    await page.waitForChanges();
+
+    const results = await page.evaluate(
+      async (): Promise<{ childTitle: string; childContent: string; parentContent: string; parentTitle: string }> => {
+        document.getElementById("kid-b").click();
+
+        await new Promise<void>((resolve) => requestAnimationFrame(() => resolve()));
+        await new Promise<void>((resolve) => requestAnimationFrame(() => resolve()));
+
+        const parentTab = document.getElementById("parent-tab");
```
Comment (jcfranco) [inline src/components/tabs/tabs.e2e.ts:301]: Can the test setup use the Puppeteer/E2E APIs or does it need to run in the browser context?

**Review by jcfranco — COMMENTED** [review]
Noice! I mainly had some comments about the test.

I really think we should try to keep refactoring separate from PRs unless truly needed by the fix/feature. Makes it simpler for the reviewer to process the changes and helps us keep commits semantic. I'll submit a separate PR to split up the clean up work.

**Comment by driskull** [comment]
> I really think we should try to keep refactoring separate from PRs unless truly needed by the fix/feature. Makes it simpler for the reviewer to process the changes and helps us keep commits semantic.

Agreed. It can be tough to get to those refactoring issues though. These were pretty simple fixes that made sense to do while in the file.

I'll try to be mindful of them and keep them separate.

**Comment by driskull** [comment]
@jcfranco fixed test. If all is good can you merge?

**Review by jcfranco — APPROVED** [review]
🎉
